crypto/encode_decode/decoder_pkey.c

```diff
@@ -148,7 +150,14 @@ static int decoder_construct_pkey(OSSL_DECODER_INSTANCE *decoder_inst,
          * result in the keymgmt.
          */
         if (keymgmt_prov == decoder_prov) {
+            ERR_set_mark();
             keydata = evp_keymgmt_load(keymgmt, object_ref, object_ref_sz);
+            if (keydata == NULL
+                && data->ctx->errkeep == NULL
+                && (data->ctx->errkeep = OSSL_ERR_STATE_new()) != NULL) {
+                OSSL_ERR_STATE_save_to_mark(data->ctx->errkeep);
+                ERR_pop_to_mark();
+            }
```

Comment:
This is not the normal pattern I see littered about the code..
Why does this not also ERR_clear_last_mark() if the ERR_pop_to_mark() is not called.

Reply:
Thanks. Added the missing clear.